### PR TITLE
Javadoc enhancement for JsonObject and JsonArray

### PR DESCRIPTION
**Pull request by EmadAlblueshi**
Signed-off-by: Emad Alblueshi <emad.albloushi@gmail.com>

### CODE REVIEW

File: src/main/java/io/vertx/core/json/JsonArray.java [hunk 1]
```diff
@@ -224,6 +225,7 @@ public JsonArray getJsonArray(int pos) {
    * @param pos  the position in the array
    * @return  the Instant, or null if a null value present
    * @throws java.lang.ClassCastException if the value cannot be converted to String
+   * @throws java.time.format.DateTimeParseException if the String value is not a legal ISODATE encoded value
```
Comment (cescoffier) [inline src/main/java/io/vertx/core/json/JsonArray.java]: Maybe we should specify the expected format (Not sure everyone knows what ISODATE is).

File: src/main/java/io/vertx/core/json/JsonObject.java [hunk 2]
```diff
@@ -259,7 +259,7 @@ public JsonArray getJsonArray(String key) {
    * @param key  the key to return the value for
    * @return the value or null if no value for that key
    * @throws java.lang.ClassCastException if the value is not a String
-   * @throws java.lang.IllegalArgumentException if the String value is not a legal Base64 encoded value
+   * @throws java.time.format.DateTimeParseException if the String value is not a legal ISODATE encoded value
```
Comment (cescoffier) [inline src/main/java/io/vertx/core/json/JsonObject.java]: Maybe we should specify the expected format (Not sure everyone knows what ISODATE is).

File: src/main/java/io/vertx/core/json/JsonArray.java (hunk 1 above)
Comment (EmadAlblueshi) [inline src/main/java/io/vertx/core/json/JsonArray.java]: I think the format example should be mentioned in the paragraph above and leave the ISODATE as is

Comment (cescoffier) [inline src/main/java/io/vertx/core/json/JsonArray.java]: Up to you for the location, it just needs to be somewhere.

**Comment by EmadAlblueshi** [comment]
@cescoffier Done

**Comment by cescoffier** [comment]
👍

**Pull request merged**